### PR TITLE
Adding fileset info & FS diagrams

### PR DESCRIPTION
**Pull request by hflynn**
To address https://trac.openmicroscopy.org.uk/ome/ticket/10793 and https://trello.com/c/zd9PQRlo/232-better-fs-docs-with-diagrams

Open to any suggestions on where to add the managed repo diagrams from http://downloads.openmicroscopy.org/presentations/2014/Paris-Workshops/OMERO-FS-Workshop/#/4 too, if anyone thinks they are useful, I'm not clear what they show without explanatory text.

### CODE REVIEW

**Comment by hflynn** [comment]
N.B. fileset text mostly taken from what @mtbc wrote some time again linked to the ticket, for credit where credit is due ;)

File: omero/developers/ImportFS.txt [hunk 1]
```diff
@@ -42,6 +46,35 @@ client all processing then takes place on the server. The server then uses
 Bio-Formats to extract and store the metadata, calculate the minimum and
 maximum pixel values and do other import processing.
 
+.. _filesets:
+
+Filesets
+--------
+
+A ``fileset`` is a concept new in OMERO.fs that captures how
```
Comment (rleigh-dundee) [inline omero/developers/ImportFS.txt]: new to ?

File: omero/developers/ImportFS.txt [hunk 2]
```diff
@@ -42,6 +46,35 @@ client all processing then takes place on the server. The server then uses
...
+
+.. figure:: /images/fileset-diagram.png
+  :align: center
+  :alt: Filesets can be many-to-one, one-to-one, one-to-many or
+        many-to-many
+
+Filesets are essentially indivisible: the files or images that a fileset
+represents are deleted, or moved between groups, only as one unit together,
+and on the server each fileset has a directory in which only its files are
+stored. Each fileset firmly binds its brittle sets of files and images because
```
Comment (rleigh-dundee) [inline omero/developers/ImportFS.txt]: Remove "its brittle" ?

File: omero/developers/ImportFS.txt [hunk 3]
```diff
@@ -42,6 +46,35 @@ client all processing then takes place on the server. The server then uses
...
+the dependencies among them mean that **splitting components away leaves a
+partial fileset and makes the remaining data unreadable**. Similarly,
```
Comment (rleigh-dundee) [inline omero/developers/ImportFS.txt]: "and makes" could be "making"

**Comment by rleigh-dundee** [comment]
All looks fine to me.  Some very minor spelling/grammar changes suggested only.  The graphics could be nicer as SVG+PDF rather than PNG, but doesn't need doing for this PR.

**Comment by hflynn** [comment]
I only have the import diagram as a png but I think I can save the fileset one in those formats, will have a play.

**Comment by hflynn** [comment]
Merging this to make sure we get the fileset docs into the tagged release. Will try to get managed repo and more import diagrams in on a further PR later this week.

**Pull request merged**